sdk/storage/azure-storage-blob/azure/storage/blob/_shared/encryption.py

```diff
@@ -115,23 +158,244 @@ def __init__(self, content_encryption_IV, encryption_agent, wrapped_content_key,
             A dict containing metadata related to the key wrapping.
         '''
 
-        _validate_not_none('content_encryption_IV', content_encryption_IV)
         _validate_not_none('encryption_agent', encryption_agent)
         _validate_not_none('wrapped_content_key', wrapped_content_key)
 
+        # Validate we have the right matching optional parameter for the specified algorithm
+        if encryption_agent.encryption_algorithm == _EncryptionAlgorithm.AES_CBC_256:
+            _validate_not_none('content_encryption_IV', content_encryption_IV)
+        elif encryption_agent.encryption_algorithm == _EncryptionAlgorithm.AES_GCM_256:
+            _validate_not_none('encrypted_region_info', encrypted_region_info)
+        else:
+            raise ValueError("Invalid encryption algorithm.")
+
         self.content_encryption_IV = content_encryption_IV
+        self.encrypted_region_info = encrypted_region_info
         self.encryption_agent = encryption_agent
         self.wrapped_content_key = wrapped_content_key
         self.key_wrapping_metadata = key_wrapping_metadata
 
 
-def _generate_encryption_data_dict(kek, cek, iv):
+class GCMBlobEncryptionStream:
+    """
+    A stream that performs AES-GCM encryption on the given data as
+    it's streamed. Data is read and encrypted in regions. The stream
+    will use the same encryption key and will generate a guaranteed unique
+    nonce for each encryption region.
+    """
+    def __init__(
+            self,
+            content_encryption_key: bytes,
+            data_stream: BinaryIO,
+            ):
+        """
+        :param content_encryption_key: The encryption key to use.
```

Comment:
Maybe because this is `bytes` and it is convention to not list `bytes` as the type -- but if not, perhaps it should be `:param bytes content_encryption_key`? Duplicate this comment for `data_stream` as well for possibly missing a paramtype?